### DISCUSSION TITLE
Multiple foreign keys pointing the same table?

### DISCUSSION THREAD

**Discussion by seed-of-apricot** [prisma/prisma#2073] [Q&A] [answered]
If I have the following data structure:

```ts
type User = {
  id: number;
  name: string;
  attitude1: Likert;
  attitude2: Likert;
  attitude3: Likert;
};

type Likert = {
  id: number;
  label: string;
  value: number;
};
```
I want to have this datamodel

```prisma
model User {
  id        Int @id
  name      String
  attitude1 Likert @relation(fields: [attitude1Id], references: [id])
  attitude1Id Int
  attitude2 Likert @relation(fields: [attitude2Id], references: [id])
  attitude2Id Int
  attitude3 Likert @relation(fields: [attitude3Id], references: [id])
  attitude3Id Int
}

model Likert {
  id    Int  @id
  label String
  value Int
  user1 User[]
  user2 User[]
  user3 User[]
}
```

But it complains

```
Error validating model "User": Ambiguous relation detected. The fields `attitude1` and `attitude2` in model `User` both refer to `Likert`. Please provide different relation names for them by adding `@relation(<name>).`.
```

How can I achieve this structure? Shouldn't I make multiple foreign keys in the first place?

**Comment by janpio** [comment]
> Please provide different relation names for them by adding `@relation(<name>

Have you tried this? 
`@relation("foo", fields: [attitude1Id], references: [id])` would be the syntax here I think.

**Reply by seed-of-apricot** [reply]
I didn't know this syntax, thank you.
Do I still need to write the `fields` property with this?

**Reply by Sytten** [reply]
You only specify the `@relation` on one model (User here) for a 1-N relation.

**Reply by seed-of-apricot** [reply]
What you mentioned is different from what explained [here](https://github.com/prisma/prisma/tree/2.0.0-beta.1) in the release. Let me try both anyways.

**Reply by seed-of-apricot** [reply]
@janpio It has partially worked, but still not enough.

First, these models:

```prisma
model User {
  id          Int @id
  name        String
  attitude1   Likert @relation("attitude1", fields: [attitude1Id], references: [id])
  attitude2   Likert @relation("attitude2", fields: [attitude2Id], references: [id])
  attitude3   Likert @relation("attitude3", fields: [attitude3Id], references: [id])
  attitude1Id Int
  attitude2Id Int
  attitude3Id Int
}

model Likert {
  id    Int  @id
  label String
  value Int
  user1 User[] 
  user2 User[] 
  user3 User[] 
}
```

causes

```
Error validating model "Likert": Ambiguous relation detected. The fields `user1` and `user2` in model `Likert` both refer to `User`. Please provide different relation names for them by adding `@relation(<name>).
```

on **user1 to user3** fields. Then, changing `Likert` to:

```prisma
model Likert {
  id    Int  @id
  label String
  value Int
  user1 User[] @relation("user1")
  user2 User[] @relation("user2")
  user3 User[] @relation("user3")
}
```

will throw 

```
Error validating model "User": Automatic related field generation would cause a naming conflict. Please add an explicit opposite relation field.

Error validating model "Likert": Automatic related field generation would cause a naming conflict. Please add an explicit opposite relation field.
```

on **`attitude2`/`attitude3`** and **`user2`/`user3`** (not on attitude1 or user1).

How can I address these issues?

**Reply by janpio** [reply]
This asks for `references` I think. Try `  user1 User[] @relation("user1", references: [attitude1])` - and I will point this discussion out to people who really know.

**Reply by janpio** [reply]
There you go: https://github.com/prisma/prisma/discussions/2073#discussioncomment-3125 Seems I misunderstood the naming a bit. so good Marcus helped out.

**Comment by mavilein** [comment, marked as answer]
You were pretty close already 😀
Ambiguous relations is an error that you get when the schema parser can not figure out which relation fields belong to the same relation. E.g. i as a human could guess that `User.attitude1` is one direction and `Likert.user1` is the other direction of the same relation. But the parser needs to understand this exactly without guessing. Through tagging the fields with the relation name the parser understands those semantics.

So in your case my guess is that you want this:

```prisma
model User {
  id        Int @id
  name      String
  attitude1 Likert @relation("attitude1", fields: [attitude1Id], references: [id])
  attitude1Id Int
  attitude2 Likert @relation("attitude2", fields: [attitude2Id], references: [id])
  attitude2Id Int
  attitude3 Likert @relation("attitude3", fields: [attitude3Id], references: [id])
  attitude3Id Int
}

model Likert {
  id    Int  @id
  label String
  value Int
  user1 User[] @relation("attitude1")
  user2 User[] @relation("attitude2")
  user3 User[] @relation("attitude3")
}
```

**Reply by seed-of-apricot** [reply]
Understood, now my linter does not complain about it, thank you!

When I have m:n relation I also need the explicit field names, right?

**Reply by DarlanMendesDigitalCollege** [reply]
Men, It actually might have saved my life <3 .I'm applying for a fullstack job and i've been spending a couple of days trying to solve this relation between my models.Thanks!!God bless u guys!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
